### DISCUSSION TITLE
ytdl的 cookies-from-browser 功能无效

### DISCUSSION THREAD

**Discussion by Lonely233233** [hooke007/mpv_PlayKit#284] [mpv 通用]
bilibili番剧现在导入cookie后无法播放了

必须在配置问价里注释掉ytdl-raw-options-append=cookies-from-browser=edge这个才能正常播放番剧
例如"https://www.bilibili.com/bangumi/play/ss4687?t=903&spm_id_from=333.1007.top_right_bar_window_history.content.click"

**Comment by zhongfly** [comment]
这应该是ytdl的问题吧，你先排除掉ytdl出问题的可能，再来mpv相关的地方问吧

**Comment by hooke007** [comment]
这里也标记一下，如果自动从浏览器获取失败那就手动指定
https://github.com/ytdl-org/youtube-dl?tab=readme-ov-file#how-do-i-pass-cookies-to-youtube-dl

懒得去翻ytdlp的issue列表了，之前看到过这个问题描述。

**Reply by hooke007** [reply]
https://github.com/yt-dlp/yt-dlp/issues/10927

**Comment by dyphire** [comment]
测试了下，发现 ytdl-raw-options-append=cookies-from-browser=Firefox 的方式依然可用，会正常使用 Firefox 的 cookies 
所以不是 cookies-from-browser 选项功能失效了，很可能是 Chromium 系的 cookies 存储方式变了，ytdlp 无法获取到浏览器的 cookies

Edit: 根据 https://meta.appinn.net/t/topic/44490/12 那里的说法：
> 另外，chrome 更新后对 cookies 文件有保护机制，你的 chrome 在打开状态下，别的程序是无法读取 Cookies 文件的，所以上面说的使用--cookies-from-browser参数需要在关闭浏览器的情况下使用，还是挺烦人的。

又测试了下 chrome 在关闭浏览器后 ytdl-raw-options-append=cookies-from-browser=Chrome 确实可以生效正常使用了。edge 不行，它默认常驻后台，必须在任务管理器中杀掉才能使 ytdl-raw-options-append=cookies-from-browser=edge 正常使用

所以这不是 ytdl 问题，是 chromium 的限制，无法解决。远离 chromium，拥抱 Firefox

**Comment by Youxikong** [comment]
yt-dlp在edge运行中无法读取edge的cookies，会报错提示文件被占用。

**Reply by Lonely233233** [reply]
"C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --disable-features=LockProfileCookieDatabase 在快捷方式后面添加一个这个就解决了

**Reply by Youxikong** [reply]
谢谢，终于可以解决了。

**Reply by dyphire** [reply]
其实还有个更通用的解决方案，为 yt-dlp 安装对应的 cookies 解锁插件：https://github.com/seproDev/yt-dlp-ChromeCookieUnlock
实测有效

**Reply by Lonely233233** [reply]
请教一下 这个怎么使用

**Reply by dyphire** [reply]
在 release 下载该插件，然后放入 yt-dlp 配置目录的插件文件夹里就可以正常生效了
文件结构参考：
![image](https://github.com/hooke007/MPV_lazy/assets/61936050/04da1cdd-78e5-47c9-b4e8-82ae449353f5)

**Reply by Lonely233233** [reply]
请教一下为什么我在mpv.conf中设置ytdl-raw-options-append=cookies-from-browser=edge无效
需要在yt-dlp.conf中设置--cookies-from-browser=edge才能正常使用读取

**Reply by dyphire** [reply]
不清楚，无法复现

**Reply by Lonely233233** [reply]
是不是mpv版本的问题 晚点我试试降低版本试试

**Reply by WeiLaiF** [reply]
> 其实还有个更通用的解决方案，为 yt-dlp 安装对应的 cookies 解锁插件：https://github.com/seproDev/yt-dlp-ChromeCookieUnlock
> 实测有效

这个好像失效了 用了几个月还正常最近用不了了，手动导出cookies正常 不知道插件还会不会更新

**Comment by verygoodlee** [comment]
最近又出新问题了，cookie用了新加密算法，基于Chromium内核的浏览器都影响到了，
尽量别更新浏览器了，我昨天更新Edge130版本已经用不了
```
[ytdl_hook] ERROR: Failed to decrypt with DPAPI. See  https://github.com/yt-dlp/yt-dlp/issues/10927  for more info
[ytdl_hook] youtube-dl failed: unexpected error occurred
```

**Reply by zhongfly** [reply]
我现在是直接把cookies全导出来用，主要是很少用yt-dlp，手动更新cookies也不麻烦

**Reply by Lonely233233** [reply]
> 最近又出新问题了，cookie用了新加密算法，基于Chromium内核的浏览器都影响到了， 尽量别更新浏览器了，我昨天更新Edge130版本已经用不了
> 
> ```
> [ytdl_hook] ERROR: Failed to decrypt with DPAPI. See  https://github.com/yt-dlp/yt-dlp/issues/10927  for more info
> [ytdl_hook] youtube-dl failed: unexpected error occurred
> ```

我今天不小心更了回退了回以前的版本了
